### PR TITLE
Add Elastic Container for Kubernetes Engine table. closes #104

### LINKED ISSUE #104: Add Elastic Container for Kubernetes Engine table.
This is to support followings

**CIS Benchmark v1.0.0 - 12-11-2020** 

7.1 Ensure Log Service is set to ‘Enabled’ on Kubernetes Engine Clusters (Automated)
7.2 Ensure CloudMonitor is set to Enabled on Kubernetes Engine Clusters (Automated)
7.3 Ensure role-based access control (RBAC) authorization is Enabled on Kubernetes Engine Clusters (Automated)
7.4 Ensure Cluster Check triggered at least once per week for Kubernetes Clusters (Automated)
7.5 Ensure Kubernetes web UI / Dashboard is not enabled (Automated)
7.6 Ensure Basic Authentication is not enabled on Kubernetes Engine Clusters (Automated)
7.7 Ensure Network policy is enabled on Kubernetes Engine Clusters (Automated)
7.8 Ensure ENI multiple IP mode support for Kubernetes Cluster (Automated)
7.9 Ensure Kubernetes Cluster is created with Private cluster enabled (Automated)

### PR DESCRIPTION
**Pull request by bigdatasourav**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```

SETUP: tests/alicloud_cs_kubernetes_cluster []

PRETEST: tests/alicloud_cs_kubernetes_cluster

TEST: tests/alicloud_cs_kubernetes_cluster
Running terraform
alicloud_vpc.named_test_resource: Creating...
alicloud_vpc.named_test_resource: Creation complete after 7s [id=vpc-0xipzt8csbjoi8teb4khu]
alicloud_vswitch.named_test_resource: Creating...
alicloud_vswitch.named_test_resource: Creation complete after 7s [id=vsw-0xi65wtmksjojen71c7jm]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Creating...
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [1m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [1m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [1m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [1m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [1m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [1m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [2m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [2m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [2m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [2m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [2m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [2m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [3m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [3m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [3m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [3m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [3m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [3m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [4m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [4m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [4m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [4m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [4m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [4m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [5m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [5m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [5m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [5m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [5m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [5m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [6m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [6m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [6m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [6m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [6m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [6m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [7m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [7m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [7m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [7m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [7m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [7m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [8m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [8m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [8m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [8m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Creation complete after 8m34s [id=c88e15807fd744fafbdbb9101c77ad276]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

cluster_id = "c88e15807fd744fafbdbb9101c77ad276"
region = "us-east-1"
resource_aka = "acs:cs:us-east-1:5982111499156037:container/turbottest22935"
resource_name = "turbottest22935"

Running SQL query: test-get-query.sql
[
  {
    "cluster_id": "c88e15807fd744fafbdbb9101c77ad276",
    "name": "turbottest22935",
    "size": 2
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "cluster_id": "c88e15807fd744fafbdbb9101c77ad276",
    "name": "turbottest22935",
    "state": "running"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "acs:cs:us-east-1:5982111499156037:container/turbottest22935"
    ],
    "name": "turbottest22935",
    "title": "turbottest22935"
  }
]
✔ PASSED

POSTTEST: tests/alicloud_cs_kubernetes_cluster

TEARDOWN: tests/alicloud_cs_kubernetes_cluster

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  name,
  cluster_id,
  state,
  size,
  cluster_type
from
  alicloud_elastic_container;
+------+-----------------------------------+---------+------+-------------------+
| name | cluster_id                        | state   | size | cluster_type      |
+------+-----------------------------------+---------+------+-------------------+
| test | cc73a722cd1274e4488df2cb8c88ee4b9 | running | 1    | ManagedKubernetes |
+------+-----------------------------------+---------+------+-------------------+
> select
  name,
  cluster_id,
  state,
  size,
  cluster_type
from
  alicloud_elastic_container
where
  state = 'running'
+------+-----------------------------------+---------+------+-------------------+
| name | cluster_id                        | state   | size | cluster_type      |
+------+-----------------------------------+---------+------+-------------------+
| test | cc73a722cd1274e4488df2cb8c88ee4b9 | running | 1    | ManagedKubernetes |
+------+-----------------------------------+---------+------+-------------------+
> select
  name,
  cluster_id,
  state,
  size,
  cluster_type
from
  alicloud_elastic_container
where
  cluster_type = 'ManagedKubernetes'
+------+-----------------------------------+---------+------+-------------------+
| name | cluster_id                        | state   | size | cluster_type      |
+------+-----------------------------------+---------+------+-------------------+
| test | cc73a722cd1274e4488df2cb8c88ee4b9 | running | 1    | ManagedKubernetes |
+------+-----------------------------------+---------+------+-------------------+

### CODE REVIEW

**Comment by bigdatasourav** [comment]
```

SETUP: tests/alicloud_elastic_container []

PRETEST: tests/alicloud_elastic_container

TEST: tests/alicloud_elastic_container
Running terraform
alicloud_vpc.named_test_resource: Creating...
alicloud_vpc.named_test_resource: Creation complete after 7s [id=vpc-0xis82rce4oojirzjc58c]
alicloud_vswitch.named_test_resource: Creating...
alicloud_vswitch.named_test_resource: Creation complete after 7s [id=vsw-0xib7x0gmd64bbm2tb8j5]
alicloud_cs_kubernetes.named_test_resource: Creating...
alicloud_cs_kubernetes.named_test_resource: Still creating... [10s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [20s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [30s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [40s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [50s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [1m0s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [1m10s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [1m20s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [1m30s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [1m40s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [1m50s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [2m0s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [2m10s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [2m20s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [2m30s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [2m40s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [2m50s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [3m0s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [3m10s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [3m20s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [3m30s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [3m40s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [3m50s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [4m0s elapsed]
alicloud_cs_kubernetes.named_test_resource: Still creating... [4m10s elapsed]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Error: [ERROR] terraform-provider-alicloud/alicloud/resource_alicloud_cs_kubernetes.go:799: Resource id：cfc41d33c442d4015b81e2beec5012565 :
[ERROR] terraform-provider-alicloud/alicloud/service_alicloud_cs.go:317:
Failed to reach target status. Current status is failed.

  on variables.tf line 41, in resource "alicloud_cs_kubernetes" "named_test_resource":
  41: resource "alicloud_cs_kubernetes" "named_test_resource" {


Terraform run failed, skipping SQL queries

✘ FAILED
Terraform run failed, skipping SQL queries

✘ FAILED
Terraform run failed, skipping SQL queries

✘ FAILED
Terraform run failed, skipping SQL queries

✘ FAILED

TEARDOWN: tests/alicloud_elastic_container

File: alicloud-test/tests/alicloud_elastic_container/variables.tf [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+data "alicloud_caller_identity" "current" {}
+
+data "null_data_source" "resource" {
+  inputs = {
+    scope = "acs:::${data.alicloud_caller_identity.current.account_id}"
+  }
+}
+
+
```
Comment (Subhajit97) [inline alicloud-test/tests/alicloud_elastic_container/variables.tf:26]:
```suggestion
```

File: alicloud-test/tests/alicloud_elastic_container/variables.tf [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  worker_vswitch_ids    = [alicloud_vswitch.named_test_resource.id, alicloud_vswitch.named_test_resource.id, alicloud_vswitch.named_test_resource.id]
+  master_instance_types = ["ecs.c5.large", "ecs.c5.large", "ecs.c5.large"]
+  worker_instance_types = ["ecs.c5.large", "ecs.c5.large", "ecs.c5.large"]
+  worker_number         = "3"
+  pod_cidr              = "10.10.0.0/24"
+  service_cidr          = "192.168.0.0/24"
+  password              = "Test1234!"
+  name                  = var.resource_name
+}
+
```
Comment (Subhajit97) [inline alicloud-test/tests/alicloud_elastic_container/variables.tf:52]:
```suggestion
```

File: alicloud-test/tests/alicloud_elastic_container/variables.tf [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+output "cluster_id" {
+  value = alicloud_cs_kubernetes.named_test_resource.id
+}
+
```
Comment (Subhajit97) [inline alicloud-test/tests/alicloud_elastic_container/variables.tf:57]:
```suggestion
```

File: alicloud/plugin.go [hunk 4]
```diff
@@ -42,6 +42,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"alicloud_vpc_vpn_customer_gateway": tableAlicloudVpcVpnCustomerGateway(ctx),
 			"alicloud_vpc_vpn_gateway":          tableAlicloudVpcVpnGateway(ctx),
 			"alicloud_vpc_vswitch":              tableAlicloudVpcVSwitch(ctx),
+			"alicloud_elastic_container":        tableAlicloudElasticContainer(ctx),
```
Comment (Subhajit97) [inline alicloud/plugin.go:45]: Can we change the table name to `alicloud_cs_kubernetes` as per terraform?
And also sort the list

File: alicloud/service.go [hunk 5]
```diff
@@ -324,3 +325,31 @@ func RDSService(ctx context.Context, d *plugin.QueryData, region string) (*rds.C
 
 	return svc, nil
 }
+
+// CSService returns the service connection for Alicloud Container service
+func CSService(ctx context.Context, d *plugin.QueryData, region string) (*cs.Client, error) {
```
Comment (Subhajit97) [inline alicloud/service.go:330]:
```suggestion
func ContainerService(ctx context.Context, d *plugin.QueryData, region string) (*cs.Client, error) {
```

File: alicloud/table_alicloud_elastic_container.go [hunk 6]
```diff
@@ -0,0 +1,481 @@
...
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/cs"
+
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAlicloudElasticContainer(ctx context.Context) *plugin.Table {
```
Comment (Subhajit97) [inline alicloud/table_alicloud_elastic_container.go:18]:
```suggestion
func tableAlicloudCSKubernetesCluster(ctx context.Context) *plugin.Table {
```

File: alicloud/table_alicloud_elastic_container.go [hunk 7]
```diff
@@ -0,0 +1,481 @@
...
+	return &plugin.Table{
+		Name:        "alicloud_elastic_container",
```
Comment (Subhajit97) [inline alicloud/table_alicloud_elastic_container.go:20]:
```suggestion
		Name:        "alicloud_cs_kubernetes_cluster",
```

File: alicloud/table_alicloud_elastic_container.go [hunk 8]
```diff
@@ -0,0 +1,481 @@
...
+				Transform:   transform.FromField("size"),
+			},
+			{
+				Name:        "cluster_type",
+				Description: "The type of the cluster.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("cluster_type"),
+			},
+			{
+				Name:        "created",
```
Comment (Subhajit97) [inline alicloud/table_alicloud_elastic_container.go:62]: Can we change the column name to `created_at`?

File: alicloud/table_alicloud_elastic_container.go [hunk 9]
```diff
@@ -0,0 +1,481 @@
...
+			{
+				Name:        "network_mode",
+				Description: "The network type of the cluster.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("network_mode"),
+			},
+			{
+				Name:        "subnet_cidr",
+				Description: "The CIDR block of pods in the cluster.",
+				Type:        proto.ColumnType_STRING,
```
Comment (Subhajit97) [inline alicloud/table_alicloud_elastic_container.go:142]:
```suggestion
				Type:        proto.ColumnType_CIDR,
```

File: alicloud/table_alicloud_elastic_container.go [hunk 10]
```diff
@@ -0,0 +1,481 @@
...
+			{
+				Name:        "vswitch_id",
+				Description: "The IDs of VSwitches.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("vswitch_id"),
+			},
+			{
+				Name:        "vswitch_cidr",
+				Description: "The CIDR block of VSwitches.",
+				Type:        proto.ColumnType_STRING,
```
Comment (Subhajit97) [inline alicloud/table_alicloud_elastic_container.go:112]:
```suggestion
				Type:        proto.ColumnType_CIDR,
```

File: alicloud/table_alicloud_elastic_container.go [hunk 11]
```diff
@@ -0,0 +1,481 @@
...
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getElasticContainerAka,
+				Transform:   transform.FromValue(),
+			},
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("name"),
+			},
```
Comment (Subhajit97) [inline alicloud/table_alicloud_elastic_container.go:308]: The order should be `title`, `tags` and `akas`

File: alicloud/table_alicloud_elastic_container.go [hunk 12]
```diff
@@ -0,0 +1,481 @@
...
+				Hydrate:     getCommonColumns,
+				Transform:   transform.FromField("AccountID"),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listElasticContainers(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
```
Comment (Subhajit97) [inline alicloud/table_alicloud_elastic_container.go:330]:
```suggestion
func listCsKubernetesClusters(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
```

File: alicloud/table_alicloud_elastic_container.go [hunk 13]
```diff
@@ -0,0 +1,481 @@
...
+			break
+		}
+		request.PageNumber = requests.NewInteger(int(PageNumber) + 1)
+	}
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getElasticContainer(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (Subhajit97) [inline alicloud/table_alicloud_elastic_container.go:373]:
```suggestion
func getCsKubernetesCluster(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

File: alicloud/table_alicloud_elastic_container.go [hunk 14]
```diff
@@ -0,0 +1,481 @@
...
+	if len(response.GetHttpContentString()) > 0 {
+		var cluster map[string]interface{}
+		json.Unmarshal([]byte(response.GetHttpContentString()), &cluster)
+		return cluster, nil
+	}
+
+	return nil, nil
+}
+
+func getClusterLog(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (Subhajit97) [inline alicloud/table_alicloud_elastic_container.go:411]:
```suggestion
func getCsKubernetesClusterLog(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

File: alicloud/table_alicloud_elastic_container.go [hunk 15]
```diff
@@ -0,0 +1,481 @@
...
+	}
+
+	if len(response.GetHttpContentString()) > 0 {
+		return response.GetHttpContentString(), nil
+	}
+
+	return nil, nil
+}
+
+func getElasticContainerAka(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (Subhajit97) [inline alicloud/table_alicloud_elastic_container.go:447]:
```suggestion
func getCsKubernetesClusterAka(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@bigdatasourav Please make suggested changes

File: alicloud/alicloud_cs_kubernetes_cluster.go [hunk 16]
```diff
@@ -0,0 +1,481 @@
...
+			},
+			{
+				Name:        "external_loadbalancer_id",
+				Description: "The ID of the Server Load Balancer (SLB) instance deployed in the cluster.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("external_loadbalancer_id"),
+			},
+			{
+				Name:      "port",
+				Type:      proto.ColumnType_STRING,
```
Comment (Subhajit97) [inline alicloud/alicloud_cs_kubernetes_cluster.go:153]: Can you please add a suitable description fro this column?

File: alicloud/alicloud_cs_kubernetes_cluster.go [hunk 17]
```diff
@@ -0,0 +1,481 @@
...
+			},
+			{
+				Name:        "docker_version",
+				Description: "The version of Docker.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("docker_version"),
+			},
+			{
+				Name:      "swarm_mode",
+				Type:      proto.ColumnType_STRING,
```
Comment (Subhajit97) [inline alicloud/alicloud_cs_kubernetes_cluster.go:176]: Can you add a description?

File: alicloud/alicloud_cs_kubernetes_cluster.go [hunk 18]
```diff
@@ -0,0 +1,481 @@
...
+			{
+				Name:      "upgrade_components",
+				Type:      proto.ColumnType_STRING,
+				Transform: transform.FromField("upgrade_components"),
+			},
+			{
+				Name:      "next_version",
+				Type:      proto.ColumnType_STRING,
+				Transform: transform.FromField("next_version"),
+			},
```
Comment (Subhajit97) [inline alicloud/alicloud_cs_kubernetes_cluster.go:193]: Can you add a description for these columns?

File: alicloud/alicloud_cs_kubernetes_cluster.go [hunk 19]
```diff
@@ -0,0 +1,481 @@
...
+				Transform:   transform.FromValue(),
+			},
+			{
+				Name:        "tags_src",
+				Description: ColumnDescriptionTags,
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("tags"),
+			},
+
+			// steampipe standard columns
```
Comment (Subhajit97) [inline alicloud/alicloud_cs_kubernetes_cluster.go:289]:
```suggestion
			// Steampipe standard columns
```

File: alicloud/alicloud_cs_kubernetes_cluster.go [hunk 20]
```diff
@@ -0,0 +1,481 @@
...
+			},
+			{
+				Name:        "akas",
+				Description: ColumnDescriptionAkas,
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getCsKubernetesClusterAka,
+				Transform:   transform.FromValue(),
+			},
+
+			// alicloud standard columns
```
Comment (Subhajit97) [inline alicloud/alicloud_cs_kubernetes_cluster.go:310]:
```suggestion
			// Alicloud standard columns
```

File: alicloud/alicloud_cs_kubernetes_cluster.go [hunk 21]
```diff
@@ -0,0 +1,481 @@
...
+			return nil, err
+		}
+		var result map[string]interface{}
+		json.Unmarshal([]byte(response.GetHttpContentString()), &result)
+		clusters := result["clusters"].([]interface{})
+		pageInfo := result["page_info"].(map[string]interface{})
+		TotalCount := pageInfo["total_count"].(float64)
+		PageNumber := pageInfo["page_number"].(float64)
+		for _, cluster := range clusters {
+			plugin.Logger(ctx).Warn("listCsKubernetesClusters", "item", cluster)
```
Comment (Subhajit97) [inline alicloud/alicloud_cs_kubernetes_cluster.go:358]:
```suggestion
```

File: alicloud/alicloud_cs_kubernetes_cluster.go [hunk 22]
```diff
@@ -0,0 +1,481 @@
...
+		return nil, err
+	}
+	commonColumnData := commonData.(*alicloudCommonColumnData)
+	accountID := commonColumnData.AccountID
+
+	akas := []string{"acs:cs:" + data["region_id"].(string) + ":" + accountID + ":container/" + data["name"].(string)}
+
+	return akas, nil
+}
+
```
Comment (Subhajit97) [inline alicloud/alicloud_cs_kubernetes_cluster.go:464]:
```suggestion

//// TRANSFORM FUNCTIONS

```

File: alicloud/plugin.go [hunk 23]
```diff
@@ -42,6 +42,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"alicloud_vpc_vpn_customer_gateway": tableAlicloudVpcVpnCustomerGateway(ctx),
 			"alicloud_vpc_vpn_gateway":          tableAlicloudVpcVpnGateway(ctx),
 			"alicloud_vpc_vswitch":              tableAlicloudVpcVSwitch(ctx),
+			"alicloud_cs_kubernetes_cluster":    tableAlicloudCsKubernetesCluster(ctx),
```
Comment (Subhajit97) [inline alicloud/plugin.go:45]: Sort the list

File: alicloud/service.go [hunk 24]
```diff
@@ -324,3 +325,31 @@ func RDSService(ctx context.Context, d *plugin.QueryData, region string) (*rds.C
...
 	return svc, nil
 }
+
+// ContainerService returns the service connection for Alicloud Container service
+func ContainerService(ctx context.Context, d *plugin.QueryData, region string) (*cs.Client, error) {
+	if region == "" {
+		return nil, fmt.Errorf("region must be passed ContainerService")
+	}
+	// have we already created and cached the service?
+	serviceCacheKey := fmt.Sprintf("ContainerService-%s", region)
```
Comment (Subhajit97) [inline alicloud/service.go:335]:
```suggestion
	serviceCacheKey := fmt.Sprintf("cs-%s", region)
```

File: docs/tables/alicloud_cs_kubernetes_cluster.md [hunk 25]
```diff
@@ -0,0 +1,48 @@
...
+  name,
+  cluster_id,
+  state,
+  size,
+  cluster_type
+from
+  alicloud_elastic_container;
+```
+
+### List of clusters which are currently running
```
Comment (Subhajit97) [inline docs/tables/alicloud_cs_kubernetes_cluster.md:20]:
```suggestion
### List clusters which are running
```

File: docs/tables/alicloud_cs_kubernetes_cluster.md [hunk 26]
```diff
@@ -0,0 +1,48 @@
...
+  state,
+  size,
+  cluster_type
+from
+  alicloud_elastic_container
+where
+  state = 'running';
+```
+
+### List of clusters which are managed kubernetes
```
Comment (Subhajit97) [inline docs/tables/alicloud_cs_kubernetes_cluster.md:35]:
```suggestion
### List clusters which are managed by kubernetes
```

File: alicloud/alicloud_cs_kubernetes_cluster.go (hunk 18 above)
Comment (bigdatasourav) [reply alicloud/alicloud_cs_kubernetes_cluster.go:193]: as discussed , proper description is not available for these columns

File: alicloud-test/tests/alicloud_cs_kubernetes_cluster/test-list-query.sql [hunk 27]
```diff
@@ -0,0 +1,3 @@
+select name, cluster_id, state
+from alicloud_cs_kubernetes_cluster
+where akas::text = '["{{ output.resource_aka.value }}"]' and region = '{{ output.region.value }}';;
```
Comment (Subhajit97) [inline alicloud-test/tests/alicloud_cs_kubernetes_cluster/test-list-query.sql:3]:
```suggestion
where akas::text = '["{{ output.resource_aka.value }}"]' and region = '{{ output.region.value }}';
```

Comment (Subhajit97) [reply alicloud-test/tests/alicloud_cs_kubernetes_cluster/test-list-query.sql:3]: @bigdatasourav Is region needed to list?

File: alicloud-test/tests/alicloud_cs_kubernetes_cluster/variables.tf [hunk 28]
```diff
@@ -0,0 +1,66 @@
...
+output "resource_name" {
+  value = var.resource_name
+}
+
+output "region" {
+  value = var.alicloud_region
+}
+
+output "resource_aka" {
+  value = "acs:cs:us-east-1:${data.alicloud_caller_identity.current.account_id}:container/${var.resource_name}"
```
Comment (Subhajit97) [inline alicloud-test/tests/alicloud_cs_kubernetes_cluster/variables.tf:65]:
```suggestion
  value = "acs:cs:${var.alicloud_region}:${data.alicloud_caller_identity.current.account_id}:container/${var.resource_name}"
```

File: alicloud-test/tests/alicloud_cs_kubernetes_cluster/test-list-query.sql (hunk 27 above)
Comment (bigdatasourav) [reply alicloud-test/tests/alicloud_cs_kubernetes_cluster/test-list-query.sql:3]: yes

File: alicloud/alicloud_cs_kubernetes_cluster.go [hunk 29]
```diff
@@ -0,0 +1,487 @@
...
+		Description: "Alicloud Container Service Kubernetes Cluster",
+		List: &plugin.ListConfig{
+			Hydrate: listCsKubernetesClusters,
+		},
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.AllColumns([]string{"cluster_id", "region"}),
+			Hydrate:    getCsKubernetesCluster,
+		},
+		GetMatrixItem: BuildRegionList,
+		Columns: []*plugin.Column{
```
Comment (Subhajit97) [inline alicloud/alicloud_cs_kubernetes_cluster.go:30]: Please arrange the column in following order:
1. Add all the important columns ( max. 4-5 columns) at the top
2. Add rest of the columns in alphabetical order
3. Add the JSON type columns at the end

File: alicloud/alicloud_cs_kubernetes_cluster.go [hunk 30]
```diff
@@ -0,0 +1,487 @@
...
+			{
+				Name:        "profile",
+				Description: "The identifier of the cluster.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("profile"),
+			},
+			{
+				Name:        "deletion_protection",
+				Description: "Indicates whether deletion protection is enabled for the cluster.",
+				Type:        proto.ColumnType_STRING,
```
Comment (Subhajit97) [inline alicloud/alicloud_cs_kubernetes_cluster.go:215]: Is the type correctly defined? Or, it should be boolean?

File: alicloud/alicloud_cs_kubernetes_cluster.go [hunk 31]
```diff
@@ -0,0 +1,487 @@
...
+			{
+				Name:        "cluster_log",
+				Description: "The logs of a cluster.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getCsKubernetesClusterLog,
+				Transform:   transform.FromValue(),
+			},
+			{
+				Name:        "tags_src",
+				Description: ColumnDescriptionTags,
```
Comment (Subhajit97) [inline alicloud/alicloud_cs_kubernetes_cluster.go:285]:
```suggestion
				Description: "A list of tags attached with the cluster.",
```

File: alicloud/alicloud_cs_kubernetes_cluster.go [hunk 32]
```diff
@@ -0,0 +1,489 @@
+package alicloud
```
Comment (Subhajit97) [inline alicloud/alicloud_cs_kubernetes_cluster.go:1]: Add the `table_` prefix in file name

File: alicloud/alicloud_cs_kubernetes_cluster.go [hunk 33]
```diff
@@ -0,0 +1,489 @@
...
+				Transform:   transform.FromField("size"),
+			},
+			{
+				Name:        "created_at",
+				Description: "The time when the cluster was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("created"),
+			},
+
+			// Other columns
```
Comment (Subhajit97) [inline alicloud/alicloud_cs_kubernetes_cluster.go:62]:
```suggestion
```

File: docs/tables/alicloud_cs_kubernetes_cluster.md [hunk 34]
```diff
@@ -0,0 +1,48 @@
+# Table: alicloud_elastic_container
```
Comment (Subhajit97) [inline docs/tables/alicloud_cs_kubernetes_cluster.md:1]:
```suggestion
# Table: alicloud_cs_kubernetes_container
```

File: docs/tables/alicloud_cs_kubernetes_cluster.md [hunk 35]
```diff
@@ -0,0 +1,48 @@
...
+
+```sql
+select
+  name,
+  cluster_id,
+  state,
+  size,
+  cluster_type
+from
+  alicloud_elastic_container;
```
Comment (Subhajit97) [inline docs/tables/alicloud_cs_kubernetes_cluster.md:17]: Change the table name

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@bigdatasourav Please update the fileName and resolve conflicts.
Also there are some changes required in docs. Please see comments, thanks!

File: docs/tables/alicloud_cs_kubernetes_cluster.md (hunk 34 above)
Comment (bigdatasourav) [reply docs/tables/alicloud_cs_kubernetes_cluster.md:1]: alicloud_cs_kubernetes_cluster

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**